lib/ipmi_backend_utils.pm

```diff
@@ -262,6 +262,18 @@ sub adjust_for_ipmi_xen {
     }
 }
 
+# Set boot from pxe via ipmitool
+sub set_pxe_boot {
+    while (1) {
+        my $stdout = ipmitool('chassis bootparam get 5');
+        last if $stdout =~ m/Boot Flag Valid[\d\D]*Force PXE/;
+        diag "setting boot device to pxe";
+        my $options = get_var('IPXE_UEFI') ? 'options=efiboot' : '';
+        ipmitool("chassis bootdev pxe ${options}");
+        sleep(3);
```

Comment:
Do double check after  ipmitool("chassis bootdev pxe ${options}") instead of sleep 3 ? For example,
```
ipmitool(sol info)
```

Reply:
It is the following 'ipmitool('chassis bootparam get 5');'  in next loop that does the check like your ipmitool(sol info). ipmitool("chassis bootdev pxe ${options}") does take a little time.

Reply:
OK.